### ISSUE TITLE
[Issue]: nonzero returns shape like input tensor and separate count, complicating index extraction

### ISSUE THREAD

**Issue by nttng207** [tenstorrent/tt-metal#25713]
### Is your feature request related to a problem? Please describe

The current behavior of ttnn.nonzero returns two tensors:
1. A tensor holding the count of nonzero elements, padded to the input tensor's shape.
2. A tensor with indices of nonzero elements, also shaped like the input tensor.

This deviates from the expected behavior seen in torch.nonzero, which directly returns a compact list of nonzero indices without redundant padding or the need for count slicing.

**Example**
Given:
`A = [8, 1, 2, 0, 4, 0] `
Expected output (similar to torch.nonzero(A)):
`[0, 1, 2, 4] `
Current ttnn.nonzero output:
 - tensor[0]: count tensor → [4, 0, 0, 0, ..., 0].
 - tensor[1]: index tensor → [0, 1, 2, 4, 0, 0, ..., 0] (same shape as input).

To extract the correct result, we must:
 1. Transfer the count tensor back to the host.
 2. Slice the index tensor accordingly.
This introduces a performance bottleneck due to unnecessary host dispatch and postprocessing.
Reproducible Code:
```
import torch
import ttnn
torch.manual_seed(0)
# Setup
device = ttnn.open_device(device_id=0)
input_shape = torch.Size([32])
torch_input_tensor = torch.ones(input_shape)
torch_input_tensor[::2] = 0  # set even-indexed elements to 0
input_tensor = ttnn.from_torch(torch_input_tensor, layout=ttnn.ROW_MAJOR_LAYOUT, device=device)
# Apply ttnn.nonzero
output_tensor = ttnn.nonzero(input_tensor, queue_id=0)
# Retrieve count (requires host dispatch)
output_tensor1 = ttnn.to_layout(output_tensor[0], ttnn.ROW_MAJOR_LAYOUT)
output_tensor1 = ttnn.from_device(output_tensor1)
output_tensor1 = ttnn.to_torch(output_tensor1)
nonzero_count = output_tensor1[..., 0].item()
# Extract actual indices
index_tensor = output_tensor[1][:nonzero_count]
```
### Desired Behavior
ttnn.nonzero should ideally mimic the PyTorch API by returning a compact tensor of only the nonzero indices (i.e., output.shape == [N] where N is the number of nonzero elements). This eliminates:
- The need for host-side slicing
- Redundant data transfer
- Extra logic to extract the result

### Describe the solution you'd like

Consider modifying the API to return only the relevant tensor with desired shape. That is, the behavior should be like `torch.nonzero` (https://docs.pytorch.org/docs/stable/generated/torch.nonzero.html).

### Describe alternatives you've considered

We understand that dynamic tensor allocation is not yet supported due to architectural constraints in `tt-metal`, which leads to the current static, preallocated output format. What can be a suggested workaround solution for now?

_No response_
